### DISCUSSION TITLE
Feature Suggestion for Webhooks

### DISCUSSION THREAD

**Discussion by HirziDevs** [pelican-dev/panel#694] [Ideas]
1. **Include Event Name in Webhook Payload**
It would be helpful to include the event name for webhooks event. Currently, only the data is sent, and it's unclear what event triggered the webhook (e.g., user updated, server created, etc.). Adding an event field like this would make it clear:
```json
{
  "event": "UserUpdated",
  "attributes": [
    {
      "id": 14,
      "uuid": "9fd89ad3-dbf8-4cdb-bb35-6ed810a73870",
      "email": "email@redacted.dev",
      "language": "en",
      "use_totp": false,
      "created_at": "2024-08-17T11:58:01.000000Z",
      "updated_at": "2024-10-29T23:08:17.000000Z",
      "name_first": null,
      "name_last": null,
      "username": "hirzidevs",
      "gravatar": true,
      "external_id": null,
      "timezone": "UTC"
    }
  ]
}
```

2. **Verification Token or API Key for Webhooks**
Could we implement a verification token or use the API key from the panel to ensure that only panel can send the webhook?

![Verification Token Example](https://github.com/user-attachments/assets/695c5041-a67c-40eb-a66c-915b105a1be2)

3. **Allow IP URL for Webhooks**
It would be great to allow IP-based URLs for webhook configurations.

**Comment by rmartinoscar** [comment]
I'v managed to obtain this
![image](https://github.com/user-attachments/assets/6306bdc3-4345-4bdb-8bdb-5e831f777403)

https://github.com/pelican-dev/panel/compare/main...RMartinOscar:panel:fix/694

**Reply by HirziDevs** [reply]
Thats already good, maybe if you can, make the event name more readable like `NODE_UPDATED` or `NodeUpdated`
